### ISSUE TITLE
createOptionForm doesn't work inside a repeater

### ISSUE THREAD

**Issue by hfallahnezhad** [filamentphp/filament#2948]
### Package

filament/forms

### Package Version

v2.13.10

### Laravel Version

v9.18.0

### Livewire Version

v2.10.5

### PHP Version

PHP 8.1.7

### Bug description

Using `Select`->`createOptionForm()` inside a `Repeater`. It adds a `+` button besides the combobox but it doesn't work.
"It works fine when you use it as a normal field, but when you put the `Select` field inside a `Repeater`, it doesn't work."

### Steps to reproduce

Create a form like this:

```php
.
.
.
  , Forms\Components\Repeater::make('shows')
	  ->relationship('shows')
	  ->label(__('Shows'))
	  ->schema([
		  Forms\Components\TextInput::make('title')
			  ->label(__('Title'))
			  ->required()
			  ->maxLength(255),
		  Forms\Components\Select::make('program_id')
			  ->label(__('Program'))
			  ->relationship('program', 'title')
			  ->createOptionForm([
				  Forms\Components\TextInput::make('title')
					  ->label(__('Title'))
					  ->required()
					  ->maxLength(255),
				  Forms\Components\Textarea::make('description')
					  ->label(__('Description'))
					  ->maxLength(65535),
		])
	])
  ]),
.
.
.
```
Then click on `+` in the form, and nothing happens

**Comment by wijnenmik** [comment]
This is also not working for me

**Comment by zepfietje** [comment]
Can you provide us a reproduction repository URL and then we'll look at your issue?

That would allow us to download it and review your bug much easier, so it can be fixed quicker. Please make sure to give us some instructions on where to find the bug, and include a database seeder with everything we need to set it up quickly.

**Comment by danharrin** [comment]
Fixed by #3413.
